pipeline_tools/http_requests.py

```diff
@@ -234,12 +234,16 @@ def _http_request(self, *args, **kwargs):
 
     @staticmethod
     def _get_method(http_method):
+        session = requests.Session()
+        # The default max_redirects is 30. We need to boost it because DSS sometimes redirects us more than 30 times,
+        # which causes an error if we don't allow for more.
+        session.max_redirects = 100
```

Comment:
These changes make sense to me. But I'm not sure why Blue is redirecting us is it based on any `User-Agent `headers? If so, how can we avoid that? Redirecting more than 30 times seems like a misbehavior to me.

Reply:
Blue box recently incorporated the checkout service into requests for the bundle manifest. This means that there is no longer a guarantee that they will instantly return the bundle manifest. Instead, they redirect us in a loop back to the same url (with some added parameters) so that when the manifest is ready we will get it. See @ttung's recently merged PR here: https://github.com/HumanCellAtlas/data-store/pull/1366. So we have to be able to handle a large number of redirects for these calls.

Reply:
Thanks for the explanation here! This makes more sense to me now.